Added AsTypeLookup and ArithmeticLookup

NdubisiOnuora opened the pull request:
@miki725, @gregarmer, can you please review?

Commit:
(1) Updated version to 0.3.1 and HISTORY.rst to indicate new improvements in 0.3.1.
(2) Added AsTypeLookup and ArithmeticLookup.
(3) Clarified use-case for ListConfig.

miki725 commented on simplepath/lookups.py:
```diff
@@ -69,3 +81,62 @@ def repr(self):
+            reverse (bool): whether the operation should be reversed;
+                            Example: for subtraction, operand - node
+        """
+        OPERATORS = {
```
maybe move this as class attribute, same as in lookup above?

Commit:
Utils
1) Added a utility for division to accomodate the different implementations between Python 3 and Python 2 given the fact that Python 3 uses operator.truediv for the "/" operator while Python 2 uses operator.div for said operator.

ArithmeticLookup
1) Replace use of operator.div with utils.divide.
2) Moved the OPERATORS variable in config to a class property.

Commit:
Lookups
1) Fixed incorrect import for utils.

Commit:
TestDivide
1) Made sure that both tests can run in environments where operator.div is not defined.

NdubisiOnuora commented:
@miki725, addressed comments and all tests have passed.

Commit:
Importanize on modified files containing modified imports.

miki725 commented on simplepath/lookups.py:
```diff
@@ -23,6 +27,16 @@ def call_expression(self, expression, data, extra):
+        and using the value in the current node.
+
+        Note:
+            When overriding this method, make sure to not modify
```
:+1:

miki725 commented on simplepath/utils.py:
```diff
@@ -50,3 +51,11 @@ def deepvars(data_object):
         return data_object
+
+
+def divide(a, b):
```
this seems unnecessary. lets rather go with Python3 style division - https://docs.python.org/3.4/library/operator.html#mapping-operators-to-functions

> ```
> Division    a / b   truediv(a, b)
> Division    a // b  floordiv(a, b)
> ```

btw, same is supported in Py2 so that should not cause any compatibility issues

miki725 replied:
so in other words, you can move this to the lookup itself and register both `/` and `//` with appropriate `operator` functions

NdubisiOnuora replied:
@miki725, I'll make the change, but the reason I did it this way is that "/" is `floordiv` in Python 2 but `truediv` in Python 3. I didn't want users in Python 2 to be surprised when "/" becomes `truediv`.

miki725 replied:
we can mention that in the lookup doc that it works like in Python3. Plus in Python2 people can do:

``` python
from __future__ import division
```

miki725 commented on tests/test_lookups.py:
```diff
@@ -164,3 +169,88 @@ def test_call(self):
+        self.arith_lookup = ArithmeticLookup()
+
+    def test_config_add(self):
+        self.arith_lookup.config('+', 15)
```
to be precise, `config` actually always gets the parameters as strings which is why you are doing `type(node)(self.operand)` in the lookup. might want to mimic same here...

miki725 commented:
@NdubisiOnuora LGTM just the minor division comment and passing strings in tests to `config`

Commit:
(1) Removed utils.divide.
(2) Used operator.truediv for "/" operator and operator.floordiv for "//" operator.

Commit:
Removed unused import in test_lookups and passed a string to arithmetic_lookup.config in tests.

NdubisiOnuora commented:
@miki725, addressed your comments.

Commit:
Removed additional line in import block in test_utils.py.

miki725 commented:
thanks @NdubisiOnuora
